### PR TITLE
General Code Clean up & POM Optimization

### PR DESCRIPTION
**Pull request by Crim**
Cleans up some of our dependencies and other cruft in the POM.    Fixes a bug in persistence adapter test where it wasn't closing out the adapter at the end of the test

### CODE REVIEW

**Comment by Crim** [comment]
Sorry about the poor branch name -- I rebased my previous branch off of master.

File: pom.xml [hunk 1]
```diff
@@ -85,9 +85,10 @@
         <kafkaVersion>0.10.0.1</kafkaVersion>
 
         <!-- Define which Curator version -->
-        <curatorVersion>2.12.0</curatorVersion>
-
+        <curatorVersion>4.0.0</curatorVersion>
```
Comment (Crim) [inline pom.xml]: This is the verison used as a zookeeper client in our ZK implementations.

File: pom.xml [hunk 2]
```diff
@@ -85,9 +85,10 @@
...
 
+        <!-- Define which Curator Test version -->
+        <curatorTestVersion>2.12.0</curatorTestVersion>
```
Comment (Crim) [inline pom.xml]: This is the internal zk test server that we use in tests.  From digging around on the curator website and in JIRA tickets, as far as I can tell this is the latest release that actually works, even though there are releases published up through 4.0.0.

Comment (stanlemon) [reply pom.xml]: What's the scoop with this in 4.0.0? This seems really weird...

Comment (Crim) [reply pom.xml]: No idea.  It bombs if you move it up to any version past 2.12.0.  I'll try to dig in more tomorrow and see if I can find out more.

**Pull request merged**